internal/services/access_flow.go

```diff
@@ -0,0 +1,476 @@
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	var dataGrantees []apono.GranteeV1
+	for _, grantee := range existingGrantees {
+		granteeIds, diagnostics := getIdentitiesIdsByNameAndType(ctx, grantee.Name.ValueString(), grantee.Type.ValueString(), availableIdentities.Data, aponoClient)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		for _, granteeId := range granteeIds {
+			dataGrantees = append(dataGrantees, apono.GranteeV1{
+				Id:   granteeId,
+				Type: grantee.Type.ValueString(),
+			})
+		}
+	}
+	data.Grantees = dataGrantees
+
+	existingApprovers := make([]models.Identity, 0, len(accessFlow.Approvers.Elements()))
+	diagnostics = accessFlow.Approvers.ElementsAs(ctx, &existingApprovers, false)
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	var dataApprovers []apono.ApproverV1
+	for _, approver := range existingApprovers {
+		approverIds, diagnostics := getIdentitiesIdsByNameAndType(ctx, approver.Name.ValueString(), approver.Type.ValueString(), availableIdentities.Data, aponoClient)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		for _, approverId := range approverIds {
+			dataApprovers = append(dataApprovers, apono.ApproverV1{
+				Id:   approverId,
+				Type: approver.Type.ValueString(),
+			})
+		}
+	}
+	data.Approvers = dataApprovers
+
+	availableIntegrations, _, err := aponoClient.IntegrationsApi.ListIntegrationsV2(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "integrations", "")
+	}
+
+	var dataIntegrationTargets []apono.AccessTargetIntegrationV1
+	for _, integrationTarget := range accessFlow.IntegrationTargets {
+		dataIntegrationTarget, diagnostics := convertIntegrationTargetToApiModel(integrationTarget, availableIntegrations.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		dataIntegrationTargets = append(dataIntegrationTargets, *dataIntegrationTarget)
+	}
+	data.IntegrationTargets = dataIntegrationTargets
+
+	if accessFlow.Settings != nil {
+		settings := apono.AccessFlowV1Settings{
+			RequireJustificationOnRequestAgain: *apono.NewNullableBool(accessFlow.Settings.RequireJustificationOnRequestAgain.ValueBoolPointer()),
+			RequireAllApprovers:                *apono.NewNullableBool(accessFlow.Settings.RequireAllApprovers.ValueBoolPointer()),
+			ApproverCannotApproveHimself:       *apono.NewNullableBool(accessFlow.Settings.ApproverCannotApproveHimself.ValueBoolPointer()),
+		}
+
+		data.Settings = *apono.NewNullableAccessFlowV1Settings(&settings)
+	}
+
+	return &data, nil
+}
+
+func ConvertToAccessFlowUpdateApiModel(ctx context.Context, aponoClient *apono.APIClient, accessFlow *models.AccessFlowModel) (*apono.UpdateAccessFlowV1, diag.Diagnostics) {
+	updateAccessFlowRequest, diagnostics := ConvertToAccessFlowUpsertApiModel(ctx, aponoClient, accessFlow)
+	if len(diagnostics) > 0 {
+		return nil, diagnostics
+	}
+
+	trigger := apono.UpdateAccessFlowV1Trigger{
```

Comment:
why specifically trigger is different type?

Reply:
because in update model the trigger model is nullable and the in the create model its required so when creating client for GO it split the Trigger to 2 different models